framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -3771,5 +3801,577 @@ std::wstring Dx12ReplayConsumerBase::ConstructObjectName(format::HandleId captur
+        }
+        track_dump_resources_.record_render_pass_flags = Flags;
+    }
+}
+
+void Dx12ReplayConsumerBase::PreCall_ID3D12GraphicsCommandList_DrawInstanced(const ApiCallInfo& call_info,
+                                                                             DxObjectInfo*      object_info,
+                                                                             UINT               VertexCountPerInstance,
+                                                                             UINT               InstanceCount,
+                                                                             UINT               StartVertexLocation,
+                                                                             UINT               StartInstanceLocation)
+{
+    if (options_.dump_resources_type != DumpResourcesType::kNone &&
+        track_dump_resources_.target.drawcall_code_index == call_info.index)
+    {
+        auto commandlist = reinterpret_cast<ID3D12GraphicsCommandList*>(object_info->object);
+        if (track_dump_resources_.target.begin_renderpass_code_index == 0)
+        {
+            // The program doesn't use renderpass.
+            AddCopyResourceCommandsForBeforeDrawcall(commandlist);
+        }
+        else
+        {
+            // Insert renderpass
+            graphics::dx12::ID3D12GraphicsCommandList4ComPtr commandlist4;
+            commandlist->QueryInterface(IID_PPV_ARGS(&commandlist4));
+            commandlist4->EndRenderPass();
+
+            AddCopyResourceCommandsForBeforeDrawcall(commandlist);
+
+            std::vector<D3D12_RENDER_PASS_RENDER_TARGET_DESC> rt_descs;
+            auto rt_descs_size = track_dump_resources_.replay_render_target_handles.size();
+            rt_descs.resize(rt_descs_size);
+            for (uint32_t i = 0; i < rt_descs_size; ++i)
+            {
+                rt_descs[i].cpuDescriptor        = track_dump_resources_.replay_render_target_handles[i];
+                rt_descs[i].BeginningAccess.Type = D3D12_RENDER_PASS_BEGINNING_ACCESS_TYPE_PRESERVE;
+                rt_descs[i].EndingAccess.Type    = D3D12_RENDER_PASS_ENDING_ACCESS_TYPE_PRESERVE;
+            }
+
+            D3D12_RENDER_PASS_DEPTH_STENCIL_DESC* p_ds_desc = nullptr;
+            D3D12_RENDER_PASS_DEPTH_STENCIL_DESC  ds_desc{};
+            if (track_dump_resources_.depth_stencil_heap_id != 0)
+            {
+                p_ds_desc                           = &ds_desc;
+                ds_desc.cpuDescriptor               = track_dump_resources_.replay_depth_stencil_handle;
+                ds_desc.DepthBeginningAccess.Type   = D3D12_RENDER_PASS_BEGINNING_ACCESS_TYPE_PRESERVE;
+                ds_desc.StencilBeginningAccess.Type = D3D12_RENDER_PASS_BEGINNING_ACCESS_TYPE_PRESERVE;
+                ds_desc.DepthEndingAccess.Type      = D3D12_RENDER_PASS_ENDING_ACCESS_TYPE_PRESERVE;
+                ds_desc.StencilEndingAccess.Type    = D3D12_RENDER_PASS_ENDING_ACCESS_TYPE_PRESERVE;
```

Comment:
Here is to insert renderpass in before drawcall. The BeginningAccess an EndingAcccess of the renderpass have to be PRESERVE to keep the the render process for this following renderpass, or it will clear the render target.